### ISSUE TITLE
Provide a fluent interface

### ISSUE THREAD

**Issue by MovGP0** [davezych/shience#8]
As a personal opinion, there should be a fluent interface like the `StringBuilder` and many modern libraries like [Polly](https://github.com/App-vNext/Polly) have. 

```csharp
Shience.SetPublisher(new FilePublisher(@"C:\file\path\to\results.txt"));

var userCanRead = Shience.New<bool>("widget-permissions")
    .Setup(() => ...)
    .Test(
        control: (() => return UserPermissions.CheckUser(currentUser); ), 
        candidate: (() => return User.Can(currentUser, Permission.Read); )
    )
    .WithComparer(new PermissionComparer())
    .When(() => if-condition)
    .WithContext(new { ... })
    .TestPercentage(0.3d)
    .PublishTo(result => ...)
    .Execute(); 

if(userCanRead)
{
    //do things!
}
```

The advantage of this is that it becomes way easier to extend and maintain the interface. Also, the code for setting up an Scientist (extensions) can be separated from the Scientist (state) into different classes. 

The user profits because the order of the properties becomes less important and the improved Auto Complete experience.

**Comment by davezych** [comment]
I agree, and is something I had been considering more and more. 

Up for grabs if you want it, otherwise I'll look into it soon.

**Comment by davezych** [comment]
I have some time today, I'll start this now...

**Comment by MovGP0** [comment]
oh. obviously we both worked on the same item...

**Comment by davezych** [comment]
Implemented in #13. Closing.
